### PR TITLE
Stubs

### PR DESCRIPTION
**Pull request by emmauss**
stubs for ns, hid,am, vi, nifm and svc:010 . makes some games boot further

### CODE REVIEW

File: Ryujinx.Core/OsHle/Svc/SvcThread.cs [hunk 1]
```diff
@@ -51,6 +51,18 @@ private void SvcStartThread(AThreadState ThreadState)
             //TODO: Error codes.
         }
 
+        private void SvcGetCurrentProcessorNumber(AThreadState ThreadState)
+        {
+            KThread CurrThread = Process.GetThread(ThreadState.Tpidr);
+
+            if (CurrThread != null)
+            {
+                ThreadState.X0 = 0;
```
Comment (gdkchan) [inline Ryujinx.Core/OsHle/Svc/SvcThread.cs]: You should just set X0 to CurrThread.ProcessorId (which is the processor id where the thread is currently running).

**Comment by emmauss** [comment]
I will be adding more stubs as the guys on discord discover them

File: Ryujinx.Core/OsHle/Svc/SvcThread.cs [hunk 2]
```diff
@@ -51,6 +51,18 @@ private void SvcStartThread(AThreadState ThreadState)
...
+            }
+
+            //TODO: Error codes.
```
[the remark this answers is not in the document]
Comment (emmauss) [reply Ryujinx.Core/OsHle/Svc/SvcThread.cs]: Like a "ThreadNullException"?

[the remark this answers is not in the document]
Comment (gdkchan) [reply Ryujinx.Core/OsHle/Svc/SvcThread.cs]: if it was me I would just remove the `CurrThread != null` check so it throws if it is null.

**Comment by emmauss** [comment]
@gdkchan  check the nullthreadexception i made. any issues

File: Ryujinx.Core/OsHle/Exceptions/NullThreadException.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace Ryujinx.Core.OsHle.Exceptions
+{
+    class NullThreadException : Exception
+    {
+        private const string ExMsg = "Thread with ThreadId {} is null!";
+
+        public NullThreadException() : base() { }
+
+        public NullThreadException(int ThreadId) : base(string.Format(ExMsg,ThreadId)) { }
```
Comment (gdkchan) [inline Ryujinx.Core/OsHle/Exceptions/NullThreadException.cs]: nit: Space after the `,`.

**Comment by gdkchan** [comment]
It's, imo, not necessary, you can just remove the check so it throws a NullReferenceException.

**Comment by emmauss** [comment]
okay

File: Ryujinx.Core/OsHle/Svc/SvcHandler.cs [hunk 4]
```diff
@@ -35,6 +35,7 @@ public SvcHandler(Switch Ns, Process Process)
                 { 0x07, SvcExitProcess                   },
                 { 0x08, SvcCreateThread                  },
                 { 0x09, SvcStartThread                   },
+                { 0x10, SvcGetCurrentProcessorNumber     },
```
Comment (gdkchan) [inline Ryujinx.Core/OsHle/Svc/SvcHandler.cs]: nit: this is 0x10 in hex, not decimal, so it should go after 0xf.

**Comment by emmauss** [comment]
squashed

File: Ryujinx.Core/OsHle/Svc/SvcThread.cs [hunk 5]
```diff
@@ -115,6 +115,14 @@ private void SvcSetThreadCoreMask(AThreadState ThreadState)
             //TODO: Error codes.
         }
 
+        private void SvcGetCurrentProcessorNumber(AThreadState ThreadState)
+        {
+            KThread CurrThread = Process.GetThread(ThreadState.Tpidr);
+
+            ThreadState.X0 = (ulong)CurrThread.ProcessorId;
+
```
Comment (LDj3SNuD) [inline Ryujinx.Core/OsHle/Svc/SvcThread.cs]: New line (@123) needed?

Comment (emmauss) [reply Ryujinx.Core/OsHle/Svc/SvcThread.cs]: no

**Comment by gdkchan** [comment]
lgtm, thanks!

**Pull request merged**